src/lxc/utils.c

```diff
@@ -2330,9 +2330,10 @@ int run_command(char *buf, size_t buf_size, int (*child_fn)(void *), void *args)
 	/* close the write-end of the pipe */
 	close(pipefd[1]);
 
-	bytes = read(pipefd[0], buf, (buf_size > 0) ? (buf_size - 1) : 0);
-	if (bytes > 0)
+	if (buf && buf_size > 0) {
+		bytes = read(pipefd[0], buf, buf_size - 1);
 		buf[bytes - 1] = '\0';
```

Comment:
If read returns -1, you'll do buf[-2] = '\0';

Reply:
Rigt, thanks for spitting this!

Reply:
<spit>

Reply:
Iiiinteresting.  Let's try that again.  What happens when I type:

"<spit>"

Reply:
<blink>Look, I'm annoying</blink>
I really didn't think comments did this sort of thing.  I'm disappointed.

Reply:
s/spitting/spotting/
Damn autocorrect on the phone.